File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        lo, hi, res = 0, len(nums) - 1, [0]*len(nums)
        for i in range(len(nums) - 1, -1, -1):
            if abs(nums[lo]) >= abs(nums[hi]):
                res[i] = (nums[lo] * nums[lo])
                lo += 1
            else:
                res[i] = (nums[hi] * nums[hi])
                hi -= 1
        return res

File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int lo = 0; int hi = nums.length - 1;
        int[] res = new int[nums.length];
        for(int i = nums.length - 1; i >= 0; i--){
            if(Math.abs(nums[lo]) >= Math.abs(nums[hi])){
                res[i] = nums[lo] * nums[lo];
                lo++;
            }
            else{
                res[i] = nums[hi] * nums[hi];
                hi--;
            }
        }
        return res;
    }
}